0014-fungible-token-standard.md

```diff
@@ -0,0 +1,98 @@
+
+The API consists of the following `TypeScript` `interface`s:
+
+### Transferable
+This interface consists of a single function, `transfer`, which sends a specified amount of tokens from one account to another.
+
+```TypeScript
+
+interface Transferable {
+  transfer(from: PublicKey | AccountUpdate,
+           to: PublicKey | AccountUpdate,
+           amount: UInt64 | number | bigint): void;
+}
+```
+
+### Approvable
+Transactions that are either more complicated than a transfer from one account to another (such as sending tokens from one account to many accounts) or that are constructed from third-party contracts can be done using the `Approvable` interface. The workflow is to construct the individual `AccountUpdate` values, and then have them be approved by the owner contract. The interface defines three functions to cover the cases of individual `AccountUpdate`s or `AccountUpdateTree`s, an array of either of these types, or a whole `AccountUpdateForest`.
+
+```TypeScript
+interface Approvable {
+
+  approveAccountUpdate(accountUpdate: AccountUpdate | AccountUpdateTree): void;
+  approveAccountUpdates(accountUpdates: (AccountUpdate | AccountUpdateTree)[]): void;
+  approveBase(forest: AccountUpdateForest): void;
+}
+```
+
+### Administrative Interfaces
+For administrative actions, such as changing the supply or updating the `VerificationKey`, we define the `Mintable`, `Burnable`, and `Upgradeable` interfaces:
+
+```TypeScript
+interface Mintable {
+  totalSupply: State<UInt64>;
+  circulatingSupply: State<UInt64>;
+  mint: (to: PublicKey, amount: UInt64) => AccountUpdate;
+  setTotalSupply: (amount: UInt64) => void;
+}
+
+interface Burnable {
+  burn: (from: PublicKey, amount: UInt64) => AccountUpdate;
+}
+
+interface Upgradable {
+  setVerificationKey: (verificationKey: VerificationKey) => void;
+}
+```
+
+### Viewable
+For retrieving information about the token, such as the total or circulating supply, account balances, etc., we define the `Viewable` interface.
+
+```TypeScript
+interface Viewable {
+  getAccountOf: (address: PublicKey) => ReturnType<typeof Account>;
+  getBalanceOf: (address: PublicKey, options: ViewableOptions) => UInt64;
+  getTotalSupply: (options: ViewableOptions) => UInt64;
+  getCirculatingSupply: (options: ViewableOptions) => UInt64;
+  getDecimals: () => UInt64;
```

Comment:
I suppose decimals will not be a on-chain state and a hard coded constant, correct?

Reply:
Right, that will be static for any token, so no need to have it as state.